File: app/src/main/java/net/yuntara/dspplayer/SongAdapter.java
package net.yuntara.dspplayer;

import android.widget.BaseAdapter;
import java.util.ArrayList;
import android.content.Context;
import android.view.LayoutInflater;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.view.View;
import android.view.ViewGroup;

/**
 * Created by yuntara on 2016/11/03.
 */

public class SongAdapter extends BaseAdapter {

    private ArrayList<Song> songs;
    private LayoutInflater songInf;
    private int selPos;
    public SongAdapter(Context c, ArrayList<Song> theSongs){
        songs=theSongs;
        songInf=LayoutInflater.from(c);
        selPos=-1;
    }

    @Override
    public int getCount() {
        return songs.size();
    }

    @Override
    public Object getItem(int arg0) {

        return null;
    }

    @Override
    public long getItemId(int arg0) {

        return 0;
    }
    public void setSelected(int position){
        selPos = position;
        this.notifyDataSetChanged();
    }
    public void refreshList(ArrayList<Song> theSongs){
        //songs.clear();
        songs = theSongs;
        this.notifyDataSetChanged();
    }
    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        //map to song layout
        LinearLayout songLay;
        if(convertView==null) {
            songLay = (LinearLayout) songInf.inflate(R.layout.song, parent, false);
        }else {
            songLay = (LinearLayout)convertView;
        }

        //get title and artist views
        TextView songView = (TextView) songLay.findViewById(R.id.song_title);
        TextView artistView = (TextView) songLay.findViewById(R.id.song_artist);
        ImageView img = (ImageView) songLay.findViewById(R.id.playingIcon);


                //get song using position
        Song currSong = songs.get(position);

        //get title and artist strings
        songView.setText(currSong.getTitle());
        artistView.setText(currSong.getArtist());
        if(position == selPos){
            img.setVisibility(View.VISIBLE);
        }else{
            img.setVisibility(View.INVISIBLE);
        }
        //set position as tag
        songLay.setTag(position);



        return songLay;
    }
}

File: app/src/main/java/net/yuntara/dspplayer/SongAdapter.kt
package net.yuntara.dspplayer

import android.widget.BaseAdapter
import java.util.ArrayList
import android.content.Context
import android.view.LayoutInflater
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import android.view.View
import android.view.ViewGroup

/**
 * Created by yuntara on 2016/11/03.
 */

class SongAdapter(c: Context, private var songs: ArrayList<Song>?) : BaseAdapter() {
    private val songInf: LayoutInflater
    private var selPos: Int = 0

    init {
        songInf = LayoutInflater.from(c)
        selPos = -1
    }

    override fun getCount(): Int {
        return songs!!.size
    }

    override fun getItem(arg0: Int): Any? {

        return null
    }

    override fun getItemId(arg0: Int): Long {

        return 0
    }

    fun setSelected(position: Int) {
        selPos = position
        this.notifyDataSetChanged()
    }

    fun refreshList(theSongs: ArrayList<Song>) {
        //songs.clear();
        songs = theSongs
        this.notifyDataSetChanged()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        //map to song layout
        val songLay: LinearLayout
        if (convertView == null) {
            songLay = songInf.inflate(R.layout.song, parent, false) as LinearLayout
        } else {
            songLay = convertView as LinearLayout
        }

        //get title and artist views
        val songView = songLay.findViewById(R.id.song_title) as TextView
        val artistView = songLay.findViewById(R.id.song_artist) as TextView
        val img = songLay.findViewById(R.id.playingIcon) as ImageView


        //get song using position
        val currSong = songs!![position]

        //get title and artist strings
        songView.text = currSong.title
        artistView.text = currSong.artist
        if (position == selPos) {
            img.visibility = View.VISIBLE
        } else {
            img.visibility = View.INVISIBLE
        }
        //set position as tag
        songLay.tag = position



        return songLay
    }
}
